ran-core/src/test/java/io/ran/ClazzMethodTest.java

```diff
@@ -0,0 +1,535 @@
+		// T[] arrayOfGeneric()
+		ClazzMethod arrayOfGeneric = Clazz.of(GenericTesterImpl.class).methods().find("arrayOfGeneric", String[].class).get();
+		assertMethod(g(GenericTester.class, String.class), arrayOfGeneric, String[].class);
+		assertTrue(arrayOfGeneric.hasGenericFromClass());
+		assertFalse(arrayOfGeneric.hasGenericFromMethod());
+	}
+
+	@Test
+	public void testArrayOfBoundGeneric() {
+		// <T2 extends T> T2[] arrayOfBoundGeneric()
+		ClazzMethod arrayOfBoundGeneric = Clazz.of(GenericTesterImpl.class).methods().find("arrayOfBoundGeneric", String[].class).get();
+		assertMethod(g(GenericTester.class, String.class), arrayOfBoundGeneric, String[].class);
+		assertFalse(arrayOfBoundGeneric.hasGenericFromClass());
+		assertTrue(arrayOfBoundGeneric.hasGenericFromMethod());
+		Clazz<?> arr = arrayOfBoundGeneric.getReturnType().getComponentType();
+		assertClazz(arr, String.class);
+	}
+
+	@Test
+	public void testArrayOfListOfT() {
+		// List<T>[] arrayOfListOfT()
+		ClazzMethod arrayOfListOfT = Clazz.of(GenericTesterImpl.class).methods().find("arrayOfListOfT", List[].class).get();
+		assertMethod(g(GenericTester.class, String.class), arrayOfListOfT, g(List[].class, String.class));
+		assertTrue(arrayOfListOfT.hasGenericFromClass());
+		assertFalse(arrayOfListOfT.hasGenericFromMethod());
+	}
+
+	@Test
+	public void arrayOfListOfWildT() {
+		// List<? extends T>[] arrayOfListOfWildT()
+		ClazzMethod arrayOfListOfWildT = Clazz.of(GenericTesterImpl.class).methods().find("arrayOfListOfWildT", List[].class).get();
+		assertMethod(g(GenericTester.class, String.class), arrayOfListOfWildT, g(List[].class, String.class));
+		assertFalse(arrayOfListOfWildT.hasGenericFromClass());
+		assertTrue(arrayOfListOfWildT.hasGenericFromMethod());
+	}
+
+	@Test
+	public void testGenericSuperWildcard() {
+		// <U extends Comparable<? super U>> NestedSelfSub orderBy(Function<String, U> sortingKeyExtractor)
+		ClazzMethod orderBy = Clazz.of(NestedSelfSub.class).methods().find("orderBy", NestedSelfSub.class, Function.class).get();
+		assertMethod(NestedSelfSub.class, orderBy,
+				/* return type */   NestedSelfSub.class,
+				/* arg 0 */            g(Function.class,
+						String.class,
+						g(Comparable.class,
+								Object.class)));
+		assertFalse(orderBy.hasGenericFromClass());
+		assertTrue(orderBy.hasGenericFromMethod());
+	}
+
+
+	@Test
+	public void testICassandraEventsQuery_ReturnCase() {
+		Clazz<?> c = Clazz.of(ICassandraEventsQuery.class);
+		ClazzMethod addMethod = c.methods().find("byMatchParentID", NestedSelf.class, Collection.class).get();
+		assertTrue(addMethod.hasGenericFromClass());
+		assertFalse(addMethod.hasGenericFromMethod());
+		Clazz<?> retType = addMethod.getReturnType();
+		assertEquals(NestedSelf.class, retType.clazz);
+		assertEquals(1, retType.generics.size());
+		assertEquals(NestedSelf.class, retType.generics.get(0).clazz);
+	}
+
+	@Test
+	public void testICassandraEventsQuery_ReturnCase_fromWildCard() {
```

Comment:
This test contains the above test by accident?

Reply:
merged them and used assertMethod